### PR TITLE
Fix IOperationSource generic type

### PR DESCRIPTION
**Pull request by isra-fel**
# Description

In management plane LRO, fix wrong inheritance of `IOperationSource<T>` where T should be a `[Resource]` instead of `[Resource]Data`.

# Checklist

To ensure a quick review and merge, please ensure:
- [x] The PR has a understandable title and description explaining the _why_ and _what_.
- [x] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [x] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Common/Generation/Writers/LongRunningOperationWriter.cs [hunk 1]
```diff
@@ -21,49 +21,6 @@ public void Write(CodeWriter writer, LongRunningOperation operation)
             var responseVariable = "response";
             var pagingResponse = operation.PagingResponse;
 
-            void WriteResultFunction(bool async)
```
Comment (isra-fel) [inline src/AutoRest.CSharp/Common/Generation/Writers/LongRunningOperationWriter.cs:24]: I refactor this nested function as a method so it is accessible to child class

File: samples/Azure.Management.Storage/Generated/StorageAccountsCreateOperation.cs [hunk 2]
```diff
@@ -57,23 +57,21 @@ internal StorageAccountsCreateOperation(ResourceOperationsBase operationsBase, C
...
+        public override ValueTask<Response<StorageAccount>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken = default) => _operation.WaitForCompletionAsync(pollingInterval, cancellationToken);
 
-        private Response<StorageAccount> MapResponseType(Response<StorageAccountData> response) => Response.FromValue(new StorageAccount(_operationBase, response.Value), response.GetRawResponse());
-
-        StorageAccountData IOperationSource<StorageAccountData>.CreateResult(Response response, CancellationToken cancellationToken)
+        StorageAccount IOperationSource<StorageAccount>.CreateResult(Response response, CancellationToken cancellationToken)
         {
             using var document = JsonDocument.Parse(response.ContentStream);
-            return StorageAccountData.DeserializeStorageAccountData(document.RootElement);
+            return new StorageAccount(_operationBase, StorageAccountData.DeserializeStorageAccountData(document.RootElement));
```
Comment (isra-fel) [inline samples/Azure.Management.Storage/Generated/StorageAccountsCreateOperation.cs:68]: Background of the PR: in mgmt-plane LRO we wish to wrap the deserialized model with a class that combines both data and operation to the resource.

**Review by chamons — CHANGES_REQUESTED** [review]
This sort of slicing with protected virtuals makes me rather uncomfortable. I'm going to start a conversation with @m-nash on better approaches forward.

**Review by chamons — APPROVED** [review]
Removing objection from landing as @m-nash is filing a tracking issue for refactoring the LRO types.

**Review by m-nash — APPROVED** [review]
Added a follow up item to track the overall design here https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5865

**Pull request merged**